### PR TITLE
PLT-2978 Added system message for changing channel purpose

### PR DESCRIPTION
**Pull request by DavidLu1997**
#### Summary
Added a system message for changing channel purpose, similar to changing channel header

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-2978

#### Checklist
- [x] Includes text changes and localization file ([.../i18n/en.json]

### CODE REVIEW

File: api/channel.go [hunk 1]
```diff
@@ -378,18 +378,54 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		}
+
+		post := &model.Post{
+			ChannelId: channelId,
+			Message:   message,
+			Type:      model.POST_PURPOSE_CHANGE,
+			UserId:    c.Session.UserId,
+			Props: model.StringInterface{
+				"old_purpose": oldChannelPurpose,
+				"new_header":  newChannelPurpose,
```
Comment (hmhealey) [inline api/channel.go]: This should be `new_purpose`. Otherwise, it looks good.

**Comment by DavidLu1997** [comment]
@hmhealey fixed

File: i18n/en.json [hunk 2]
```diff
@@ -320,6 +320,26 @@
     "translation": "%s updated the channel header to: %s"
   },
   {
+    "id": "api.channel.post_update_channel_purpose_message_and_forget.join_leave.error",
+    "translation": "Failed to post join/leave message %v"
```
Comment (grundleborg) [inline i18n/en.json]: I'm not sure if I'm misunderstanding the changes in `channel.go` above but shouldn't this error be "Failed to post channel purpose message %v"?

Comment (hmhealey) [inline i18n/en.json]: Yeah, that message isn't right. This should also be renamed since the "join_leave" part of the name doesn't make sense here

Comment (DavidLu1997) [inline i18n/en.json]: This existed above with channel header, I thought it was some special case. I'll change both.

File: i18n/en.json [hunk 3]
```diff
@@ -320,6 +320,26 @@
     "translation": "%s updated the channel header to: %s"
   },
   {
+    "id": "api.channel.post_update_channel_purpose_message_and_forget.join_leave.error",
```
Comment (grundleborg) [inline i18n/en.json]: Sorry to be pedantic, but could you rename this error to replace the `join_leave.error` part with something more specific to this situation, e.g. `create_post.error`. You can also do the same change to the header error you tidied up above. Other than that, this PR looks fine to me.

File: api/channel.go [hunk 4]
```diff
@@ -382,18 +382,54 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
...
+			message = fmt.Sprintf(utils.T("api.channel.post_update_channel_purpose_message_and_forget.updated_from"), user.Username, oldChannelPurpose, newChannelPurpose)
+		}
+
+		post := &model.Post{
+			ChannelId: channelId,
+			Message:   message,
+			Type:      model.POST_PURPOSE_CHANGE,
+			UserId:    c.Session.UserId,
+			Props: model.StringInterface{
+				"old_purpose": oldChannelPurpose,
```
Comment (coreyhulen) [inline api/channel.go]: These may not fit?  what happens when its too big?  Can we add a unit test for this?

**Pull request closed without merging**